src/mon/NVMeofGwSerialize.h

```diff
@@ -824,22 +850,41 @@ inline void decode(BeaconListener& ls, ceph::buffer::list::const_iterator &bl) {
   DECODE_FINISH(bl);
 }
 
-inline void encode(const BeaconSubsystem& sub,  ceph::bufferlist &bl) {
-  ENCODE_START(1, 1, bl);
+inline void encode(const BeaconSubsystem& sub,  ceph::bufferlist &bl, uint64_t features) {
+  uint8_t version = BEACON_VERSION_LEGACY;  // Default to legacy version
```

Comment:
If you want to use a common pair of version symbols (BEACON_VERSION_LEGACY and BEACON_VERSION_ENHANCED) for these data structures, that's reasonable, but please carefully audit all of the relevant data structures and make sure you aren't using 1 or 2 to check versions.  Please also list the data structures those versions apply to where the two symbols are defined.  I disagree with @ronen-fr 's request that you move them out of this file -- the message header needs to include this file anyway, and those two symbols are closely tied to the data structures in this file.  This file is therefore a good place for them.

Alternately, you can drop them and simply let the individual encode/decode pairs use bare, unrelated version numbers as in the rest of the code base.